pkg/cloud/aws/services/userdata/controlplane.go

```diff
@@ -154,17 +144,66 @@ type ControlPlaneInput struct {
 type ContolPlaneJoinInput struct {
 	baseUserData
 
-	CACertHash     string
-	CACert         string
-	CAKey          string
-	BootstrapToken string
-	ELBAddress     string
-	KubeConfig     string
+	CACertHash       string
+	CACert           string
+	CAKey            string
+	EtcdCACert       string
+	EtcdCAKey        string
+	FrontProxyCACert string
+	FrontProxyCAKey  string
+	SaCert           string
+	SaKey            string
+	BootstrapToken   string
+	ELBAddress       string
+}
+
+func (cpi *ControlPlaneInput) isCertMaterialExists() error {
```

Comment:
I'm confused about the name of this method, what do you mean by `isCertMaterialExists`? Same below.

Reply:
`isCertMaterialExists` verifies that the *CACert and *CAKey, for each keypair, exist.
You don't want to create userdata where cert exists but the key is empty.

Reply:
Wdyt about renaming it to something simpler like validateCertificates?

Reply:
it is not really validating certificates. it just tells you whether both cert and key are present.

Reply:
+1 to renaming, the current naming is confusing.

Reply:
OK. if it is confusing two people then I should rename it :)
Renamed.